docs/rfcs/0004-rfc-captive-portal-port-53.md

```diff
@@ -0,0 +1,39 @@
+
+## Motivation
+
+This RFC outlines the feasibility of revising the current captive portal mechanism. The captive portal IP is obtained from Guardian before the tunnel is established and is excluded from the tunnel. Additionally the VPN currently undergoes periodic captive portal checks, even when users are not authenticated. 
+
+Our current approach is inefficient and leads to potential security vulnerabilities such as [VPN-5007](https://mozilla-hub.atlassian.net/browse/VPN-5007). In regions where state and nation wide surveillance is common, organizations can setup an address which is the same as the captive portal DNS and monitor the traffic. These organizations can then fish the user into going to their deceptive URL that bypasses captive portal. While the user will eventually be redirected back to the captive portal landing page, an attacker can still verify traffic and respond with a malicious payload that can still compromise user security.
+
+The key considerations are as follows:
+
+- **Conditional Captive Portal Checks:** Captive portal checks should be performed when the user activates or deactivates VPN.
+- **Elimination of Captive Portal Checks:** If upon activation we detect a captive portal, fallback to port 53 to try and bypass the captive portal. This may permit users to bypass captive portals without sharing personal information.
+
+
+## Proposed Solution
+
+### 3.1. Conditional Captive Portal Checks
+
+  - **If Captive Portal Detected Upon Activation:** If a captive portal is detected, attempt to connect using port 53 to bypass the captive portal page.
+
+  - **No Signal Condition While VPN is Activated:** If the VPN is activated and experiences network connectivity issues ("No Signal"), show a widget to inform the user about possible diagnostics such as captive portal detection.
+
+  - **Captive Portal During Active VPN:** If user switches to a network that has captive portal while the VPN is already active (e.g., transitioning from cellular to WiFi), enter "No Signal" mode. The issue will only be resolved when the user disconnects.
+
+### 3.2. Removal of IP Exclusions
+
+  - Remove all IP address exclusions to enhance security. This logic is located in the [ConnectionManager::getExcludedAddresses()](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/connectionmanager.cpp#L558) code. This change prevents any malicious user from setting up a fake endpoint to interact with others on the network.
+
+### 3.4. Discontinuation of Heartbeat Captive Portal Checks
+
+  - Discontinue continuous captive portal checks after the tunnel is established. This allows the operating system to handle captive portal as necessary. The operating system is better equipped to handle captive portal given their integration with Network Management and automatic redirects.
```

Comment:
An example of the user experience with the operating system's handling of the captive portal may be useful to understand the change for the user.